classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -467,6 +475,23 @@ function getUnsubscribeNotificationUrl($request, $notification) {
 
 		return $unsubscribeUrl;
 	}
+
+	/**
+	 * Returns all the notification types that the user can unsubscribe from.
+	 * @return array
+	 */
+	function getUnsubscribeEnabledNotificationTypes() {
+		return array(
+			NOTIFICATION_TYPE_NEW_ANNOUNCEMENT,
+			NOTIFICATION_TYPE_SUBMISSION_SUBMITTED,
+			NOTIFICATION_TYPE_EDITOR_ASSIGNMENT_REQUIRED,
+			NOTIFICATION_TYPE_METADATA_MODIFIED,
+			NOTIFICATION_TYPE_NEW_QUERY,
+			NOTIFICATION_TYPE_QUERY_ACTIVITY,
+			NOTIFICATION_TYPE_REVIEWER_COMMENT,
+			NOTIFICATION_TYPE_PUBLISHED_ISSUE,
+		);
+	}
```

Comment:
This duplicates the list in `PKPNotificationSettingsForm::getNotificationSettingsMap()`. Perhaps we can find a common place to store this list? You'll need the list when you implement the form to unsubscribe from specific messages.

Reply:
@NateWr it seems that OJS and OMP are adding more NOTIFICATION_TYPEs along with `PKPNotificationSettingsForm::getNotificationSettingsMap()`. So the implemented approach (using the `NOTIFICATION_TYPE_PUBLISHED_ISSUE` in the lib-pkp) is wrong anyway. I made some changes. Added that list to the NotificationManager instead, and calling it from there like `NotificationManager ::getNotificationSettingsMap()`